src/flask_pyoidc/provider_configuration.py

```diff
@@ -164,22 +181,45 @@ def ensure_provider_metadata(self):
     def registered_client_metadata(self):
         return self._client_metadata
 
-    def register_client(self, redirect_uris, extra_parameters=None):
+    def register_client(self, client: Client, redirect_uri: str):
+
         if not self._client_metadata:
-            if 'registration_endpoint' not in self._provider_metadata:
+            if not self._provider_metadata['registration_endpoint']:
                 raise ValueError("Can't use dynamic client registration, provider metadata is missing "
                                  "'registration_endpoint'.")
 
             registration_request = self._client_registration_info.to_dict()
-            registration_request['redirect_uris'] = redirect_uris
-            if extra_parameters:
-                registration_request.update(extra_parameters)
-
-            resp = self.requests_session \
-                .post(self._provider_metadata['registration_endpoint'],
-                      json=registration_request,
-                      timeout=self.DEFAULT_REQUEST_TIMEOUT)
-            self._client_metadata = ClientMetadata(redirect_uris=redirect_uris, **resp.json())
+            # Check if the user has passed list of redirect_uris in ClientRegistrationInfo.
+            if 'redirect_uris' not in registration_request:
+                # If not, create it.
+                registration_request['redirect_uris'] = [redirect_uri]
+            # if it's passed, check if it's type list.
+            elif not isinstance(registration_request['redirect_uris'], list):
+                raise ValueError('redirect_uris must be a list, not str.')
+            else:
+                # Add redirect_uri into the list of redirect_uris passed by
+                # the user in ClientRegistrationInfo.
+                registration_request['redirect_uris'].append(redirect_uri)
+            post_logout_redirect_uris = []
+            # Check if post_logout_redirect_uris exists.
+            if 'post_logout_redirect_uris' in registration_request:
+                post_logout_redirect_uris = registration_request['post_logout_redirect_uris']
+                # If it exists, add this into the list of redirect_uris.
+                registration_request['redirect_uris'].extend(post_logout_redirect_uris)
```

Comment:
Why is it needed to duplicate `post_logout_redirect_uris` in `redirect_uri` list?
Could it be something provider specific?

Reply:
IdP only understands `redirect_uris` argument. So all post logout redirect uris are supposed to be in redirect_uris for IdP to register them. Otherwise it causes invalid redirect uri on logout.

Reply:
Keycloak also does. But it doesn't add URLs from post logout redirect uris in the list of valid redirect uris during client registration. That can be the behaviour with other IdPs also. In order to ensure that IdPs always register all post logout redirect uris, we have to add it in redirect_uris.